src/main/java/com/stal111/forbidden_arcanus/common/loot/DynamicItemLootModifier.java

```diff
@@ -0,0 +1,46 @@
+package com.stal111.forbidden_arcanus.common.loot;
+
+import com.google.common.base.Suppliers;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import it.unimi.dsi.fastutil.objects.ObjectArrayList;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.storage.loot.LootContext;
+import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
+import net.minecraftforge.common.loot.IGlobalLootModifier;
+import net.minecraftforge.common.loot.LootModifier;
+import net.minecraftforge.registries.ForgeRegistries;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.Supplier;
+
+public class DynamicItemLootModifier extends LootModifier {
+    public static final Supplier<Codec<DynamicItemLootModifier>> CODEC = Suppliers.memoize(()
+            -> RecordCodecBuilder.create(inst -> codecStart(inst).and(ForgeRegistries.ITEMS.getCodec()
+            .fieldOf("item").forGetter(m -> m.item)).apply(inst, DynamicItemLootModifier::new)));
+    private final Item item;
+
+    public DynamicItemLootModifier(LootItemCondition[] conditionsIn, Item item) {
+        super(conditionsIn);
+        this.item = item;
+    }
+
+    @Override
+    protected @NotNull ObjectArrayList<ItemStack> doApply(ObjectArrayList<ItemStack> generatedLoot, LootContext context) {
+        for(LootItemCondition condition : this.conditions) {
```

Comment:
the conditions are tested by the public apply variant before the protected one is called

Reply:
So the for loop and if statement are redundant then?